docs/templates.md

```diff
@@ -162,6 +162,23 @@ resource "docker_container" "workspace" {
 }
 ```
 
+#### Using updated images when rebuilding a workspace
+
+To ensure that Coder uses an updated image when rebuilding a workspace, set the
+`imagePullPolicy` to `Always` in your Terraform template; when set, Coder will
+check `image:tag` on every build and update if necessary:
+
+```tf
+resource "kubernetes_pod" "podName" {
+    spec {
+        container {
+            image_pull_policy = "Always"
+        }
+    }
+}
+```
```

Comment:
This leads to a significantly slower "start" time, even if the image tag hasn't been updated. Alternatively, we can suggest that admins update the tag in the template `my-image:v0.4.2` -> `my-image:v0.4.3` or digest `my-image:[digest]` -> `my-image:[digest]`

Alternatively, the `docker_image` data source can be used to scan a registry for the latest digest (I haven't tested this).

```hcl
variable "image_name" {
   default = "ubuntu:precise"
}


data "docker_registry_image" "ubuntu" {
  name = var.image_name
}

resource "kubernetes_pod" "podName" {
    spec {
        container {
            image = "${var.image_name}@sha256:${data.ubuntu.sha256digest}"
        }
    }
}
```

I can test out the second part and put it in an example, but I suggest we can recommend manually changing the tag/digest as an alternative to improve start speeds

Reply:
@bpmct do you think it would be nice to have both, explaining the pros/cons of both? I can see some places being ok with longer times if they don't have to remember to update the template

Reply:
Yeah definitely think we can have both.